src/components/checkbox/checkbox.tsx

```diff
@@ -83,8 +83,14 @@ export class Checkbox {
     private mdcCheckbox: MDCCheckbox;
     private id: string = createRandomString();
 
+    @Watch('checked')
+    protected handleCheckedChange(newValue: boolean) {
+        this.mdcCheckbox.checked = newValue;
+    }
+
     @Watch('indeterminate')
-    protected onIndeterminateChanged(newValue: boolean) {
+    protected handleIndeterminateChange(newValue: boolean) {
+        this.mdcCheckbox.checked = this.checked;
```

Comment:
```suggestion
        this.mdcCheckbox.checked = this.checked || newValue;
```

Shouldn't we force checked here if we're only allowing the interaction indeterminate => unchecked (and never indeterminate => checked)?

Reply:
We do allow indeterminate -> checked, if indeterminate is unset programmatically.

If the consumer changes the value of `indeterminate` back and forth, without changing the value of `checked`, I didn't want to put `this.mdcCheckbox.checked` in a state that doesn't match `this.checked`, for fear of causing incorrect transitions somewhere else.